### ISSUE TITLE
razorpay checkout not working in reactnative project.

### ISSUE THREAD

**Issue by sandeep3690Butte** [razorpay/react-native-razorpay#277]
Hello,

I had an reactnative project created with expo.I eject this project so that we can integrate our app with razorpay payment gateway.Done all the changes mentioned on the razorypay official doc for reactnative integration.

while loading the razorpay checkout on our app the chekout is not loading completely and is navigating directly back to our homescreen. I even won't see any errors on the remote debugger.

Any help here is really been appreciated as we are trying to rool-out our beta app  for next week and are waiting on this  to be completed soon.

Platform Details:
"react-native-razorpay": "^2.1.31",
"react-native": "~0.61.2",
"react": "~16.9.0",

**Comment by anasmp4** [comment]
i have same issue here

**Comment by tomkaos** [comment]
I have the same issue! Anybody has a solution?

"react": "16.9.0",
"react-native": "0.61.4",
"react-native-razorpay": "^2.1.35",

**Comment by globefire** [comment]
@Nautiyalsachin or any other contributor can you please suggest a solution for this?

PS: I get an error from MainApplicationJava saying that the package is getting overridden by native_module.

**Comment by globefire** [comment]
@sandeep3690Butte @anasmp4 

Check out this, this is working superbly fine for me.
https://github.com/PallaviDutta/RazorPayIntegration-ReactNative

**Comment by HrushikeshAgrawal** [comment]
A downgrade to react-native-razorpay@2.1.33 helped me solve the issue.
Can confirm that the problem exists in com.razorpay:checkout:1.5.18 updated from 1.5.12 about 10 days ago.
Would suggest everyone to uninstall latest version (npm uninstall react-native-razorpay) and install 2.1.33 (npm i react-native-razorpay@2.1.33) until the fix is offical.

**Comment by tomkaos** [comment]
> A downgrade to react-native-razorpay@2.1.33 helped me solve the issue.
> 
> Can confirm that the problem exists in com.razorpay:checkout:1.5.18 updated from 1.5.12 about 10 days ago.
> 
> Would suggest everyone to uninstall latest version (npm uninstall react-native-razorpay) and install 2.1.33 (npm i react-native-razorpay@2.1.33) until the fix is offical.

Thanks!! I will have a look

**Comment by sumedht** [comment]
@sandeep3690Butte @anasmp4 @globefire @HrushikeshAgrawal 

We have figured out the issue and it will be fixed in the next release. In the meantime, we would suggest that you don't use the previous versions and use the newer one instead as it has new features.
Add these two lines to app-level build.gradle


> implementation 'com.google.android.gms:play-services-auth:17.0.0'
> implementation 'com.google.android.gms:play-services-auth-api-phone:17.1.0'

and it should start working without a hitch. We'll also be adding this to the documentation. Please post here if you face any further issues.

**Comment by sandeep3690Butte** [comment]
Thanks a lot.I'll give it a try.

**Comment by Karthiksheety** [comment]
@sumedht 

even adding 2 line in build.gradel i am facing same issue.
checkout.open crashing my app

npm version -- "react-native-razorpay": "^2.2.0"

android config --
       buildToolsVersion = "28.0.3"
       minSdkVersion = 19
        compileSdkVersion = 28
        targetSdkVersion = 28

added these 2 lines in app-level build.gradel file 
implementation 'com.google.android.gms:play-services-auth:17.0.0'
implementation 'com.google.android.gms:play-services-auth-api-phone:17.1.0'

**Comment by vivekshindhe** [comment]
Closing this issue due to no activity. Please feel free to either re-open the issue or create a new issue if you have any additional queries.

**Comment by Sunkanmi3** [comment]
After implimenting everything am still getting this error  below
 calling this RazorpayCheckout.open() 
TypeError: Cannot read property 'open' of null

**Comment by architverma001** [comment]
I am using expo latest version and android eemilator.Also razorpay latest version 2.3.0.Still getting this issue.Please fix it

**Comment by zcodingio** [comment]
> I am using expo 49.something-- latest one ans also therazorpay latest version 2.3.0.Still getting this issue. #error [TypeError: Cannot read property 'open' of null] Please fix it. My package.son { "name": "amazonclone", "version": "1.0.0", "main": "node_modules/expo/AppEntry.js", "scripts": { "start": "expo start", "android": "expo run:android", "ios": "expo run:ios", "web": "expo start --web" }, "dependencies": { "@expo/vector-icons": "^13.0.0", "@react-native-async-storage/async-storage": "1.18.2", "@react-native-picker/picker": "^2.4.10", "@react-navigation/bottom-tabs": "^6.5.8", "@react-navigation/native": "^6.1.7", "@react-navigation/native-stack": "^6.9.13", "@reduxjs/toolkit": "^1.9.5", "axios": "^1.4.0", "expo": "~49.0.6", "expo-font": "~11.4.0", "expo-status-bar": "~1.6.0", "jwt-decode": "^3.1.2", "lottie-react-native": "^6.2.0", "react": "18.2.0", "react-native": "0.72.3", "react-native-css-vh-vw": "^1.0.5", "react-native-dropdown-picker": "^5.4.6", "react-native-image-slider-box": "^2.0.7", "react-native-modals": "^0.22.3", "react-native-razorpay": "^2.3.0", "react-native-safe-area-context": "4.6.3", "react-native-screens": "~3.22.0", "react-redux": "^8.1.2", "expo-splash-screen": "~0.20.5" }, "devDependencies": { "@babel/core": "^7.20.0" }, "private": true }

@architverma001 Did you find the issue and able to get it fixed for an Expo react native project? I am also facing the same issue.

Please help me if you were able to find the solution.

**Comment by vivekshindhe** [comment]
@zcodingio We have seen that this issue happens either when the prebuild command isn't run or when the app is running in Expo Go mode. Can you try with first running the command

npx expo prebuild 

And then, running the build in development mode?

**Comment by rahulnikam2002** [comment]
> @zcodingio We have seen that this issue happens either when the prebuild command isn't run or when the app is running in Expo Go mode. Can you try with first running the command
> 
> npx expo prebuild 
> 
> And then, running the build in development mode? 

But after building .apk it doesn't works in mobile after installing.. do uh guys have any solution?
